tests/lib/rules/forbid-component-props.js

```diff
@@ -143,11 +196,181 @@ ruleTester.run('forbid-component-props', rule, {
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', 'style'], defaultAllowOnDOM: false}],
+    errors: [
+      {
+        message: DOM_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', allow: ['div']}], defaultAllowOnDOM: false}],
+    errors: [
+      {
+        message: DOM_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', forbid: ['i']}]}],
+    errors: [
+      {
+        message: TAG_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo className="foo"><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', forbid: ['i']}]}],
+    errors: [
+      {
+        message: CLASSNAME_ERROR_MESSAGE,
+        line: 4,
+        column: 17,
+        type: 'JSXAttribute'
+      }, {
+        message: TAG_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 36,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo className="foo"><i className="foo" /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: [{property: 'className', forbid: ['i']}], defaultAllowOnComponents: true}],
+    errors: [
+      {
+        message: TAG_CLASSNAME_ERROR_MESSAGE,
+        line: 4,
+        column: 36,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
```

Comment:
If you want me to, that's cool and I'm down to learn how, but I don't know... so can you point me in the right direction if there's another plugin I might be able to read?  If not, I'll just RTFM I guess.

And do you have any problem if I do it in a separate pull request?  It seems outside the scope of this one and I might not have time to get to it until this upcoming weekend.